src/time/time.cr

```diff
@@ -567,42 +585,55 @@ struct Time
   # Time.local_offset_in_minutes # => -180
   # ```
   def self.local_offset_in_minutes
-    ifdef linux
-      -LibC.timezone.to_i32 / 60
-    else
-      if LibC.gettimeofday(nil, out tzp) != 0
-        raise Errno.new("gettimeofday")
-      end
-      -tzp.tz_minuteswest.to_i32
+    compute_ticks do |t, offset|
+      offset / Span::TicksPerMinute
     end
   end
 
   protected def self.compute_utc_ticks(ticks)
-    compute_ticks do |t, tp, tz|
-      ticks + tz
+    compute_ticks do |t, offset|
+      ticks - offset
     end
   end
 
   protected def self.compute_local_ticks(ticks)
-    compute_ticks do |t, tp, tz|
-      ticks - tz
+    compute_ticks do |t, offset|
+      ticks + offset
     end
   end
 
+  # Yields `ticks, offset`, where `ticks` is the number of ticks
+  # for "now", and `offset` is the number of ticks for now's timezone offset.
   private def self.compute_ticks
-    if LibC.gettimeofday(out tp, out tzp) != 0
-      raise Errno.new("gettimeofday")
-    end
-    ticks = tp.tv_sec.to_i64 * Span::TicksPerSecond + tp.tv_usec.to_i64 * 10_i64
-    ticks += UnixEpoch
+    LibC.tzset
 
     ifdef linux
```

Comment:
Darwin is the only POSIX platforms that still doesn't implement `clock_gettime`, this is why I went with `ifdef darwin` instead of `ifdef linux`. FreeBSD, Cygwin and other BSD variants all have it.